src/CompositeContainer.php

```
@@ -18,12 +19,15 @@ class CompositeContainer implements ContainerInterface
      */
     private $containers = [];
 
-    public function get($id)
+    public function get($id, array $parameters = [])
     {
         foreach ($this->containers as $container) {
             try {
+                if ($parameters !== []) {
+                    return $container->get($id, $parameters);
+                }
                 return $container->get($id);
-            } catch (NotFoundExceptionInterface $e) {
+            } catch (ContainerExceptionInterface $e) {
```

Should we ignore errors when instantiating existing dependencies?

---

I think we should, why not?

---

For example, sources are container1 and container2. Both have A dependency. When we're trying to retrieve A from composite container (container1, container2), out intent is clearly to get dependency from container1. It will fail because of incorrect definition, dependency will be obtained from container 2 and we'll never know why.